src/providers/mssql/qgsmssqldataitemguiprovider.cpp

```diff
@@ -171,6 +171,7 @@ void QgsMssqlDataItemGuiProvider::editConnection( QgsDataItem *item )
   {
     // the parent should be updated
     item->parent()->refreshConnections();
+    item->refresh();
```

Comment:
Is this necessary? I thought that refreshing the parent should also refresh the children but I might be wrong.

Reply:
no, I checked again, and refreshing parent was not enough